File: 1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        ret = 0
        freq = {}
        for i in nums:
            if i in freq:
                freq[i] +=1
            else:
                freq[i] =1
        for i in range(len(nums)):
            curr = nums[i]
            targ = k-curr
            if freq.get(targ,0)>0 and freq.get(curr,0) >0:
                if targ == curr and freq[curr]<2:
                    continue
                freq[curr] -=1
                freq[targ]-=1
                ret+=1
        return ret

File: 1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.ts
function maxOperations(nums: number[], k: number): number {
    let ret = 0;
    let l = 0;
    let r = nums.length-1;
    nums.sort((a,b)=> a-b)
    while (l<r){
        let curr = nums[l]+nums[r]
        if (curr==k){
            l++;
            r--;
            ret++;
        }
        else if (curr>k){
            r--;
        }
        else{
            l++;
        }
    }
    return ret;

};